src/String/Normalize.elm

```diff
@@ -40,12 +41,20 @@ removeDiacritics : String -> String
 removeDiacritics str =
     let
         replace c result =
-            case Dict.get c Diacritics.lookupTable of
-                Just candidate ->
-                    result ++ candidate
+            if Char.toCode c < Diacritics.minCode then
```

Comment:
ASCII characters have a lower Unicode code point than the lowest diacritic value. This is a cheap check that avoids the more expensive array lookup for these characters.

Reply:
Very clever optimization